auslib/db.py

```diff
@@ -1087,11 +1094,27 @@ def localeExists(self, name, platform, locale, transaction=None):
             return False
 
     def deleteRelease(self, changed_by, name, old_data_version, transaction=None):
+        self._proceedIfNotReadOnly(name, transaction=transaction)
         where = [self.name == name]
         self.delete(changed_by=changed_by, where=where, old_data_version=old_data_version, transaction=transaction)
         cache.invalidate("blob", name)
         cache.invalidate("blob_version", name)
 
+    def isReadOnly(self, name, limit=None, transaction=None):
+        where = [self.name == name]
+        column = [self.read_only]
+        row = self.select(where=where, columns=column, limit=limit, transaction=transaction)[0]
+        return row['read_only']
+
+    def _proceedIfNotReadOnly(self, name, limit=None, transaction=None):
+        if self.isReadOnly(name, limit, transaction):
+            raise ReadOnlyError("Release '%s' is read only" % name)
+
+    def changeReadOnly(self, name, read_only, changed_by, old_data_version=None, transaction=None):
+        where = [self.name == name]
+        what = dict(read_only=read_only)
+        self.update(where=where, what=what, changed_by=changed_by, old_data_version=old_data_version, transaction=transaction)
```

Comment:
Did you implement this as its own method because it would be hard to enforce the read-only condition if we supporting updating read_only in updateRelease?